scripts/rebase_patches/0020-klog-component-names.patch

```diff
@@ -0,0 +1,57 @@
+From 56e7ca38133a9f6c5edc3778e27308be9f8d397c Mon Sep 17 00:00:00 2001
+From: Ben Luddy <bluddy@redhat.com>
+Date: Wed, 31 Aug 2022 17:36:35 -0400
+Subject: [PATCH] Upsetting hack to tag klog with embedded component names.
+
+---
+ vendor/k8s.io/klog/v2/goroutine_labels.go | 24 ++++++++++++++++++++++++
+ vendor/k8s.io/klog/v2/klog.go             |  2 ++
+ 2 files changed, 26 insertions(+)
+ create mode 100644 goroutine_labels.go
+
+diff --git a/vendor/k8s.io/klog/v2/goroutine_labels.go b/vendor/k8s.io/klog/v2/goroutine_labels.go
+new file mode 100644
+index 0000000..8b28f89
+--- /dev/null
++++ b/vendor/k8s.io/klog/v2/goroutine_labels.go
+@@ -0,0 +1,24 @@
++package klog
++
++import (
++	"context"
++	"runtime/pprof"
++	"unsafe"
++)
++
++//go:linkname runtime_getProfLabel runtime/pprof.runtime_getProfLabel
++func runtime_getProfLabel() unsafe.Pointer
++
++func getMicroshiftLoggerComponent() string {
++	labels := (*map[string]string)(runtime_getProfLabel())
++	if labels != nil {
++		return (*labels)["microshift_logger_component"]
```

Comment:
Do any of the components themselves start separate goroutines that wouldn't inherit the label? Should this access check if the key is present and return a "???" if it is not?

Reply:
Since these labels are passed from spawner to spawnee, that should be OK. I think your suggestion is still useful for cases where MicroShift might be using Kubernetes packages directly / outside the scope of an embedded component.

Reply:
Done.